lib/kube/proxy/resource_deletecollection.go

```diff
@@ -495,17 +476,36 @@ func (f *Forwarder) handleDeleteCollectionReq(req *http.Request, sess *clusterSe
 	case *extensionsv1beta1.ReplicaSetList:
 		items, err := deleteResources(
 			params,
-			types.KindKubeReplicaSet,
+			sess.apiResource.resourceKind,
 			sess.apiResource.apiGroup,
+			o.APIVersion,
 			slices.ToPointers(o.Items),
-			func(ctx context.Context, client kubernetes.Interface, name, namespace string) error {
-				return trace.Wrap(client.ExtensionsV1beta1().ReplicaSets(namespace).Delete(ctx, name, deleteOptions))
-			},
+			deleteOptions,
 		)
 		if err != nil {
 			return internalErrStatus, trace.Wrap(err)
 		}
 		o.Items = slices.FromPointers(items)
+	case *unstructured.Unstructured:
+		if !o.IsList() {
+			return internalErrStatus, trace.BadParameter("unexpected CRD type")
+		}
+		list, err := o.ToList()
+		if err != nil {
+			return internalErrStatus, trace.Wrap(err)
+		}
+		items, err := deleteResources(
+			params,
+			sess.apiResource.resourceKind,
+			sess.apiResource.apiGroup,
+			o.GetAPIVersion(),
+			slices.ToPointers(list.Items),
+			deleteOptions,
+		)
+		if err != nil {
+			return internalErrStatus, trace.Wrap(err)
+		}
+		list.Items = slices.FromPointers(items)
```

Comment:
this will leak resources users don't have access to.

Reply:
How so? It is the same logic as all other types, what I am missing?

Reply:
The others are correct because we mutate the output object `o`, in this case, list is not the output object. `list` is internal to the select case